src/protocol/resp/src/request/incr.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use super::*;
+use logger::klog;
+use std::io::{Error, ErrorKind};
+use std::sync::Arc;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Incr {
+    key: Arc<[u8]>,
+}
+
+impl TryFrom<Message> for Incr {
+    type Error = Error;
+
+    fn try_from(other: Message) -> Result<Self, Error> {
```

Comment:
we should maybe make helper methods for these repetitive parsing methods. Like this one is identical to the parsing for `get`

Reply:
should definitely hold off on this while there are potential parser changes